models/user.go

```diff
@@ -432,6 +432,26 @@ func (u *User) IsPasswordSet() bool {
 	return len(u.Passwd) != 0
 }
 
+// IsVisibleToUser check if viewer is able to see user profile
+func (u *User) IsVisibleToUser(viewer *User) bool {
+	switch u.Visibility {
+	case structs.VisibleTypePublic:
+		return true
+	case structs.VisibleTypeLimited:
+		if viewer == nil || viewer.IsRestricted {
+			return false
+		}
+		return true
+	case structs.VisibleTypePrivate:
+		if viewer == nil || viewer.IsRestricted {
+			return false
+		}
+		// if private user if follow the viewer he know each other ...
+		return IsFollowing(u.ID, viewer.ID) || viewer.IsAdmin
```

Comment:
Is it legitimate? Isn't here org membership must be checked?

Reply:
Its about seeing a private user... it whuld be hidden completly, but follow ine so he can see you is a nice idear isnt it?

Reply:
Still not enough. Tried to roundup that.